File: program/find-sum-of-cubes-of-numbers-by-recursion/FindSumOfCubesOfNumbersByRecursion.kt
import kotlin.math.pow

fun main() {
    print("Enter the numbers:")
    val nums = readln()
    print("Sum of Cubes of Number: ${sumOfCubes(nums)}")
}

private fun sumOfCubes(nums: String): Int {
    val list = mutableListOf<Int>()
    var str = ""
    nums.forEach {
        if (it == ' ') {
            list.add(str.toInt())
            str = ""
        } else str += it
    }
    if (str != "") list.add(str.toInt())
    return sumRecursion(0, list)
}

private fun sumRecursion(index: Int, nums: MutableList<Int>): Int {
    if (index == nums.lastIndex) return nums[index].toDouble().pow(3).toInt()

    return nums[index].toDouble().pow(3).toInt() + sumRecursion(index + 1, nums)
}

File: program/find-sum-of-cubes-of-numbers-by-recursion/FindSumOfCubesOfNumbersByRecursion.js
/**
 * find sum of cubes of numbers by recursion
 * @param  {Number} num The first number
 *
 * @return {Number}      The sum of cubes
 */

const SumOfCubesOfNumbersByRecursion = (num) => {
  let sumOfCubesOfNumber = 0;
  if (num > 0) {
    sumOfCubesOfNumber =
      Math.pow(num % 10, 3) +
      SumOfCubesOfNumbersByRecursion(Math.floor(num / 10));
  }
  return sumOfCubesOfNumber;
};

console.log(SumOfCubesOfNumbersByRecursion(12345));
